File: Algorithms/746. Min Cost Climbing Stairs/golang/solution.bottom-up.go
func minCostClimbingStairs(cost []int) int {
	var downOne int = 0
	var downTwo int = 0
	for i := 2; i <= len(cost); i++ {
		var temp int = downOne
		if downOne+cost[i-1] < downTwo+cost[i-2] {
			downOne += cost[i-1]
		} else {
			downOne = downTwo + cost[i-2]
		}
		downTwo = temp
	}
	return downOne
}

File: Algorithms/746. Min Cost Climbing Stairs/swift/solution.bottom-up.swift
class Solution {
    func minCostClimbingStairs(_ cost: [Int]) -> Int {
        var downOne: Int = 0
        var downTwo: Int = 0
        for i in 2...cost.count {
            let temp: Int = downOne
            downOne = min(downOne + cost[i - 1], downTwo + cost[i - 2])
            downTwo = temp
        }
        return downOne
    }
}
